src/mlpack/core/data/imputation_methods/median_imputation.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file median_imputation.hpp
+ * @author Keon Kim
+ *
+ * Definition and Implementation of the MedianImputation class.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEDIAN_IMPUTATION_HPP
+#define MLPACK_CORE_DATA_IMPUTE_STRATEGIES_MEDIAN_IMPUTATION_HPP
+
+#include <mlpack/core.hpp>
+
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple median imputation
+ * replace missing value with middle or average of middle values
+ */
+template <typename T>
+class MedianImputation
+{
+ public:
+  void Impute(const arma::Mat<T>& input,
+              arma::Mat<T>& output,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool transpose = true)
+  {
+    //initiate output
+    output = input;
+
+    if (transpose)
+    {
+      arma::Mat<T> medianMat = arma::median(input, 1);
```

Comment:
Doesn't this calculate the median with the missing values (mapped values) in it?  I think we need to manually calculate the median here, ignoring the missing values.  It seems like this is not a problem for the mean imputation.